pkg/ccl/backupccl/backup_job.go

```diff
@@ -860,6 +869,29 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
 		if reloadBackupErr != nil {
 			return errors.Wrap(reloadBackupErr, "could not reload backup manifest when retrying")
 		}
+		// Re-load the job in order to update our progress object, which
+		// may have been updated since the flow started.
+		reloadedJob, reloadErr := p.ExecCfg().JobRegistry.LoadClaimedJob(ctx, b.job.ID())
+		if reloadErr != nil {
+			if ctx.Err() != nil {
+				return ctx.Err()
+			}
+			log.Warningf(ctx, "BACKUP job %d could not reload job progress when retrying: %+v",
+				b.job.ID(), reloadErr)
+		} else {
+			curProgress := reloadedJob.FractionCompleted()
+			var retryForTest bool
+			if p.ExecCfg().TestingKnobs.BeforeBackupResetRetry != nil {
+				retryForTest = p.ExecCfg().TestingKnobs.BeforeBackupResetRetry()
```

Comment:
so this retryForTest testing knob is essentially removing  testing coverage around the logic you've added for checking current and previous progress. Could you try removing this testing knob (in a new commit, to prevent erasing the good work you've already done), and adding a few more coordination mechanisms to the unit test? Check out how `TestRestoreJobRetryReset` has done this.